azavea/temperate
# WeatherEventRank endpoint

## Overview

Returns ranked WeatherEvents for the authenticated user via RegionalRiskRank (now renamed) to WeatherEventRisk.  Pulls location to check from the user's organization.

### Demo

![screen shot 2017-11-17 at 09 24 02](https://user-images.githubusercontent.com/1818302/32951748-1258248e-cb79-11e7-8082-a0eacf98ef1b.png)


### Notes

Django migrations seems to have detected that the model name change was a rename, so you shouldn't have to reload the ranked weather events via fixture. You could do it anyways to test that the fixture still works (after migrations) via `WeatherEventRank.objects.all().delete()` in a Django shell, then `./scripts/manage loaddata weathereventrank`

Some additional notes as comments in PR.


## Testing Instructions

Run migrations.

Ensure the user you will authenticate with is associated with an Organization and a location:
```
u = PlanItUser.objects.get(email='youremail')
u.get_current_location()
# Should not return None
# If it does:
org = PlanItOrganization.objects.first()
org.location = PlanItLocation.objects.from_api_city(7)
org.save()
u.organizations.add(org)
```

Then hit: http://localhost:8100/api/weather-event-rank/

Closes #169
